### PR TITLE
Add RWMutext to allow for concurrent requests

### PR DESCRIPTION
**Pull request by dougnukem**
Allow concurrent requests to go-vcr to avoid `-race` issues when running tests.

### CODE REVIEW

File: cassette/cassette.go [hunk 1]
```diff
@@ -111,6 +112,7 @@ type Cassette struct {
 	// Cassette format version
 	Version int `yaml:"version"`
 
+	lock sync.RWMutex
```
Comment (dnaeon) [inline cassette/cassette.go]: Would be better if this one is embedded.

**Comment by dougnukem** [comment]
@dnaeon updated based on your coment, but the travis-ci `tip` build issue seems to be something happening on `master`

https://travis-ci.org/dougnukem/go-vcr/jobs/169133837

**Comment by dnaeon** [comment]
Thanks, I'm on phone now and cannot merge this one, but will do so later today.

I'll check the failed build later as well.

Thanks!

**Comment by dnaeon** [comment]
Merged, thanks!

**Pull request merged**